src/libsyntax/parse/parser.rs

```diff
@@ -1532,9 +1532,8 @@ impl<'a> Parser<'a> {
         } else {
             let span = self.last_span;
             self.span_err(span,
-                          "bare raw pointers are no longer allowed, you should \
-                           likely use `*mut T`, but otherwise `*T` is now \
-                           known as `*const T`");
+                          "bare raw pointers are not allowed, use `*mut T` or \
+                           `*const T` as appropriate");
```

Comment:
Great sentiment. Can we change the wording even more. It doesn't make sense to say "bare raw pointer" when that doesn't exist as a concept (?). Technically, it's just "expected const or mut in raw pointer type (use `*mut T` or `*const T` as appropriate)", something like that.